istioctl/cmd/dashboard.go

```diff
@@ -451,7 +451,7 @@ func dashboard() *cobra.Command {
 		"Address to listen on. Only accepts IP address or localhost as a value. "+
 			"When localhost is supplied, istioctl will try to bind on both 127.0.0.1 and ::1 "+
 			"and will fail if neither of these address are available to bind.")
-	dashboardCmd.PersistentFlags().StringVar(&addonNamespace, "namespace", istioNamespace,
+	dashboardCmd.PersistentFlags().StringVarP(&addonNamespace, "namespace", "n", istioNamespace,
```

Comment:
You have to also remove `namespace` from `root.go`, so that it will be available in `istioctl dashboard -h`.
https://github.com/istio/istio/blob/18b3d35de0523862a389553b50b13ee4e73f4530/istioctl/cmd/root.go#L256